crates/cheatcodes/src/inspector.rs

```diff
@@ -1465,6 +1478,26 @@ impl Cheatcodes {
         }
     }
 
+    /// Generates arbitrary values for storage slots.
+    #[cold]
+    fn ensure_arbitrary_storage<DB: DatabaseExt>(
+        &mut self,
+        interpreter: &mut Interpreter,
+        ecx: &mut EvmContext<DB>,
+    ) {
+        if interpreter.current_opcode() == op::SLOAD {
+            let key = try_or_return!(interpreter.stack().peek(0));
+            let target_address = interpreter.contract().target_address;
+            if let Ok(value) = ecx.sload(target_address, key) {
+                if value.is_cold && value.data.is_zero() {
```

Comment:
hmm, I think this would never be cold because revm marks slot as warm during execution of opcode?

Reply:
though it looks like tests are passing? would need to look closer

Reply:
I think it might be because we're reading the value instead of a key in `interpreter.stack().peek(0)`. on `step_end` stack already contains read value

Reply:
yes, that's my take too, the opcode is not yet executed / slot marked as warm at this point

Reply:
ah so `interpreter.current_opcode()` returns _next_ opcode in `step_end`, so it's fine

Reply:
> yes, that's my take too, the opcode is not yet executed / slot marked as warm at this point

I was thinking that we're entering `ensure_arbitrary_storage` _after_ revm executed the opcode, so it's my bad

though can we document what's happening here anyway? especially that this would happen before sload even though we are in `_end` hook

Reply:
sure, will add comment to make this clear